src/common/footer_eeprom.hpp

```diff
@@ -106,7 +106,7 @@ uint8_t GetCenterNAndFewer();
  * @return constexpr uint32_t
  */
 constexpr uint32_t Encode(record rec) {
-    uint32_t ret = uint32_t(rec[0]);
+    uint32_t ret = uint32_t(rec[0]) << count_of_trailing_ones;
     for (size_t i = 1; i < count; ++i) {
```

Comment:
You can change it so that it's `uint32_t ret{0};` and then init the loop index at 0, which would result in a slightly cleaner code